packages/client/test/integration/SubscribeAll.test.ts

```diff
@@ -0,0 +1,155 @@
+        publishTestMessages = getPublishTestMessages(client, stream)
+    })
+
+    afterEach(async () => {
+        client.debug('after test')
+        expect(await M.count()).toBe(0)
+        expect(await M.count(stream.id)).toBe(0)
+        expect(M.countSubscriptionSessions()).toBe(0)
+    })
+
+    afterEach(async () => {
+        await wait(0)
+        // ensure no unexpected errors
+        expect(onError).toHaveBeenCalledTimes(expectErrors)
+    })
+
+    it('subscribes to all partitions', async () => {
+        const subMsgs: any[] = []
+        await client.subscribeAll(stream.id, (msg) => {
+            subMsgs.push(msg)
+        })
+        const eachPartition = Array(PARTITIONS).fill(0).map((_v, streamPartition) => streamPartition)
+        const pubs = await Promise.all(eachPartition.map((streamPartition) => {
+            return publishTestMessages(NUM_MESSAGES, { partitionKey: streamPartition })
+        }))
+        const publishedMsgs = pubs.flat()
+        expect(publishedMsgs.length).toBe(PARTITIONS * NUM_MESSAGES)
+        await waitForCondition(() => subMsgs.length >= (PARTITIONS * NUM_MESSAGES), 25000)
+        for (const msg of publishedMsgs) {
+            expect(subMsgs).toContainEqual(msg)
+        }
+        await client.unsubscribeAll()
+        expect(client.countAll()).toBe(0)
+    })
+
+    it('can stop prematurely', async () => {
+        const subMsgs: any[] = []
+        const sub = await client.subscribeAll(stream.id, (msg) => {
+            subMsgs.push(msg)
+            if (subMsgs.length === MAX_MESSAGES) {
+                sub.return()
+            }
+        })
+        const eachPartition = Array(PARTITIONS).fill(0).map((_v, streamPartition) => streamPartition)
+        const pubs = await Promise.all(eachPartition.map((streamPartition) => {
+            return publishTestMessages(NUM_MESSAGES, { partitionKey: streamPartition })
+        }))
+        const publishedMsgs = pubs.flat()
+        expect(publishedMsgs.length).toBe(PARTITIONS * NUM_MESSAGES)
+        await sub.onFinally()
+        // got the messages
+        expect(subMsgs).toHaveLength(MAX_MESSAGES)
+        // unsubscribed from everything
+        expect(client.countAll()).toBe(0)
+    })
+
+    it('stops with unsubscribeAll', async () => {
+        const subMsgs: any[] = []
+        const sub = await client.subscribeAll(stream.id, (msg) => {
+            subMsgs.push(msg)
+            if (subMsgs.length === MAX_MESSAGES) {
+                client.unsubscribeAll()
+            }
+        })
+        const eachPartition = Array(PARTITIONS).fill(0).map((_v, streamPartition) => streamPartition)
+        const pubs = await Promise.all(eachPartition.map((streamPartition) => {
+            return publishTestMessages(NUM_MESSAGES, { partitionKey: streamPartition })
+        }))
+        const publishedMsgs = pubs.flat()
+        expect(publishedMsgs.length).toBe(PARTITIONS * NUM_MESSAGES)
+        await sub.onFinally()
```

Comment:
What's the function of `sub.onFinally()` here?

Reply:
waits for subscription to finish, but doesn't tell it to finish.